### PR TITLE
[SPARK-5068][SQL]fix bug query data when path doesn't exists

### PR DESCRIPTION
**Pull request by jeanlyn**
the issue is descript on [SPARK-5068](https://issues.apache.org/jira/browse/SPARK-5068)
the purpose of this pull request is to prevent to make RDD for the path which doesn't exists

### CODE REVIEW

**Comment by jeanlyn** [comment]
hi, @marmbrus ,can you please take a look and give some suggestions?thx.

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala [hunk 1]
```diff
@@ -141,7 +141,11 @@ class HadoopTableReader(
       partitionToDeserializer: Map[HivePartition,
       Class[_ <: Deserializer]],
       filterOpt: Option[PathFilter]): RDD[Row] = {
-    val hivePartitionRDDs = partitionToDeserializer.map { case (partition, partDeserializer) =>
+    val hivePartitionRDDs = partitionToDeserializer.filter{ case (partition, partDeserializer) =>
```
Comment (marmbrus) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala]: Space before `{`

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala [hunk 2]
```diff
@@ -141,7 +141,11 @@ class HadoopTableReader(
...
+      val partPath = HiveShim.getDataLocationPath(partition)
+      val fs = partPath.getFileSystem(sc.hiveconf)
+      fs.exists(partPath)
```
Comment (marmbrus) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala]: This is a pretty expensive serial check, especially when there are lots of partitions.

**Comment by marmbrus** [comment]
What is the rational behind this change?  It seems like the table is corrupted and you should know about it.  Does hive work in this case?

**Comment by jeanlyn** [comment]
Yes,hive is work in this situation.I found this issue from our production environment when i try to use spark-sql to test some sql which run in hive original. I am not familiar with the business logic,but i think we should strengthen the compatibility of spark.Thanks for your check.

**Comment by marmbrus** [comment]
Okay, that is reasonable and we should probably support this. So then the question is can we do this check on the executor in parallel (or just catch the exception if it is thrown) instead of doing it serially when constructing the RDD?

**Comment by jeanlyn** [comment]
Thanks for suggestion! I would optimize this and commit later.

**Comment by marmbrus** [comment]
ok to test

**Comment by srowen** [comment]
Is this superseded by https://github.com/apache/spark/pull/3907 or https://github.com/apache/spark/pull/4356 ? if so can this be closed?

**Comment by jeanlyn** [comment]
OK.I close this one

**Pull request closed without merging**